generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity.service.ts

```diff
@@ -80,7 +80,7 @@ export class <%= entityAngularName %>Service {
         });
     }
 
-    find(id: number): Observable<<%= entityAngularName %>> {
+    find(id: string|number): Observable<<%= entityAngularName %>> {
```

Comment:
This is better to be handled with ejs so that people dont get confused so you need to do 
`<% if (databaseType == 'sql') { %>string<% } else { %>number<% } %>`